frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -93,16 +96,27 @@ function ActionCreator({
     canSave,
     isDirty,
     ui: UIProps,
-    handleActionChange,
-    handleFormSettingsChange,
+    patchAction,
+    patchFormSettings,
+    setAction,
     renderEditorBody,
   } = useActionContext();
 
-  useBeforeUnload(isDirty);
-
   const [isSaveModalShown, setShowSaveModal] = useState(false);
 
   const isEditable = isNew || (model != null && model.canWriteActions());
+  const [actionToSubmit, scheduleSubmitAction] = useState<WritebackAction>();
+
+  useEffect(() => {
```

Comment:
Another way to accomplish this is to do something like `await new Promise((resolve) => setTimeout(resolve, 0))` before calling `onClose` and `onSubmit`.

Please let me know if you think it's better that way.

I chose an effect to keep the logic and explanatory comment in one place.